File: output/deepseek_r1/avatar/direct_translation/java/python/failed_translation/atcoder_ABC168_C.py
import math

A, B, H, M = map(int, input().split())
angle_min = M * 6.0
angle_hour = H * 30 + M * 0.5
angle_diff = abs(angle_min - angle_hour)
angle_rad = math.radians(angle_diff)
distance = math.sqrt(A**2 + B**2 - 2*A*B*math.cos(angle_rad))
print(distance)

File: output/deepseek_r1/avatar/direct_translation/python/java/atcoder_ABC168_C.java

import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC168_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] parts = scanner.nextLine().split(" ");
        int a = Integer.parseInt(parts[0]);
        int b = Integer.parseInt(parts[1]);
        int h = Integer.parseInt(parts[2]);
        int m = Integer.parseInt(parts[3]);
        
        double C = Math.abs(360.0 / 60 * m - 360.0 / 12 * h - 360.0 / (12 * 60) * m);
        if (C > 180) {
            C = 360 - C;
        }
        double cosC = Math.cos(Math.toRadians(C));
        double result = Math.sqrt(a * a + b * b - 2 * a * b * cosC);
        System.out.println(result);
    }
}

